app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
For gems across components, that one hard to doc, or maybe we doc something
in every module's description ?

For gems only for one feature, like our case here, I think we can directly
put comments before the function from gem.

On Thu, Jan 8, 2015 at 6:02 PM, Joel Low <notifications@github.com> wrote:

> In app/models/instance.rb
> <https://github.com/Coursemology/coursemology2/pull/41#discussion-diff-22643413>
> :
>
> > @@ -15,6 +15,9 @@ def find_tenant_by_host(host)
> >      end
> >    end
> >
> > +  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
> > +  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
>
> that naturally begs the question: where?
>
> if you put it in your code, then you'll have the same comment appear in
> every single file... which is a lot of clutter.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Coursemology/coursemology2/pull/41/files#r22643413>.
>